kong/core/handler.lua

```diff
@@ -37,7 +38,30 @@ return {
   access = {
     before = function()
       ngx.ctx.KONG_ACCESS_START = get_now()
-      ngx.ctx.api, ngx.ctx.upstream_url, ngx.var.upstream_host = resolver.execute(ngx.var.request_uri, ngx.req.get_headers())
+      local upstream_host, balancer_address, upstream_table
+      ngx.ctx.api, ngx.ctx.upstream_url, upstream_host, upstream_table = resolve(ngx.var.request_uri, ngx.req.get_headers())
```

Comment:
We need to rename the resolver to "router". Things are becoming too confusing if we use the same word for both route matching and DNS resolution/balancing. Let's not worry about it now but in the new API matching branch I have locally I will rename it.